src/kvstore/Part.cpp

```diff
@@ -239,39 +257,80 @@ bool Part::commitLogs(std::unique_ptr<LogIterator> iter) {
             auto range = decodeMultiValues(log);
             DCHECK_EQ(2, range.size());
             if (batch->removeRange(range[0], range[1]) != ResultCode::SUCCEEDED) {
-                LOG(ERROR) << "Failed to call WriteBatch::removeRange()";
+                LOG(ERROR) << idStr_ << "Failed to call WriteBatch::removeRange()";
                 return false;
             }
             break;
         }
+        case OP_ADD_PEER:
         case OP_ADD_LEARNER: {
             break;
         }
         case OP_TRANS_LEADER: {
-            auto newLeader = decodeTransLeader(log);
+            auto newLeader = decodeHost(OP_TRANS_LEADER, log);
```

Comment:
I do not understand what is the benefit about convert string to folly::StringPiece object?

Reply:
Good catch.  const string& will be better.

Reply:
You are right, the code has been fold by git, I think the this log is the log in Part::asyncRemovePeer